packages/plugin-react/types/bugsnag-plugin-react.d.ts

```diff
@@ -1,11 +1,22 @@
-import { Plugin, OnErrorCallback } from '@bugsnag/core'
 import React from 'react'
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-interface BugsnagPluginReact extends Plugin { }
-// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+type OnErrorCallback = (event: any, cb: (err: null | Error, shouldSend?: boolean) => void) => void | boolean | Promise<void | boolean>
```

Comment:
Bit of a rabbit hole here – `OnErrorCallback` needs `@bugsnag/core/event.d.ts` which pulls in virtually everything else from core. Typed as any, but this means `onError` callbacks added on the ErrorBoundary are awkward to write.

Reply:
Yeah, this isn't great 😕 Maybe this approach isn't worth doing then? I imagine we'll hit similar issues in other plugins as well

The one saving grace is `noImplicitAny` seems to paper over this problem because you have to type the `event` parameter manually, both before & after this change